tests/test_z3_visitor.py

```diff
@@ -149,6 +159,25 @@ def not_in_empty_tuple(x: int):
 expected_list = [arithmetic_expected, boolean_expected, container_expected]
 
 
+def test_z3_vars():
+    """
+    Test the z3 variables
+    """
+    mod = astroid.parse(z3_vars_example)
+    node = mod.body[0]
+
+    cfg = ControlFlowGraph()
+    cfg.add_arguments(node.args)
+
+    # Note that this assert implicitly includes the assertion that 'a' not in cfg._z3_vars
+    assert all(
```

Comment:
This test can be improved by making the assertions more specific.

First, you should have a separate assertion for each parameter, and that assertion should also assert the *type* of z3 variable (so not just `z3.ExprRef`).

Second, assert that the size of the dictionary is 3. In general size is a good thing to check for to ensure no extraneous data was added.